docs/rfcs/035-physical-replication.md

```diff
@@ -0,0 +1,197 @@
+be running, and when it is, the replicas don't incur any extra
+overhead to the primary (see hot standby feedback though).
+
+### In-progress transactions
+
+In PostgreSQL, when a hot standby server starts up, it cannot
+immediately open up for queries. It first needs to establish a
+complete list of in-progress transactions, including subtransactions,
+that are running at the primary, at the current replay LSN. Normally
+that happens quickly, when the replica sees a "running-xacts" or
+shutdown checkpoint WAL record, because the primary writes a
+running-xacts record at every checkpoint, and in PostgreSQL the
+replica always starts the WAL replay from a checkpoint REDO point. If
+there are a lot of subtransactions in progress, however, the standby
+might need to wait for old transactions to complete before it can open
+up for queries.
+
+In Neon that problem is worse: a replica can start at any LSN, so
+there's no guarantee that it will see a running-xacts record any time
+soon. In particular, if the primary is not running when the replica is
+started, it might never see a running-xacts record.
+
+To make things worse, we initially missed this issue, and always
+started accepting queries at replica startup, even if it didn't have
+the transaction information. That could lead to incorrect query
+results and data corruption later. However, as we fixed that, we
+introduced a new problem compared to what we had before: previously
+the replica would always start up, but after fixing that bug, it migh
+tnot. In a superficious way, the old behavior was better (but could
+lead to serious issues later!). That made fixing that bug was very
+hard, because as we fixed it, we made things (superficially) worse for
+others.
+
+See https://github.com/neondatabase/neon/pull/7288 which fixed the
+bug, and follow-up PRs https://github.com/neondatabase/neon/pull/8323
+and https://github.com/neondatabase/neon/pull/8484 to try to claw back
+the cases that started to cause trouble as fixing it. As of this
+writing, there are still cases where a replica might not immediately
+start up, causing the control plane operation to fail.
+
+One long-term fix for this is to switch to using so-called CSN
+snapshots in read replica. That would make it unnecessarily to have
+the full in-progress transaction list in the replica at startup
+time. See https://commitfest.postgresql.org/48/4912/ for a
+work-in-progress patch to upstream to implement that.
+
+### Recovery conflicts and Hot standby feedback
+
+It's possible that a tuple version is vacuumed away in the primary,
+even though it is still needed by a running transactions in the
+replica. This is called a "recovery conflict", and PostgreSQL provides
+various options for dealing with it. By default, the WAL replay will
+wait up to 30 s for the conflicting query finish. After that, it will
+kill the running query, so that the WAL replay can proceed.
+
+Another way to avoid the situation is to enable `hot_standby_feedback`
```

Comment:
"**the** `hot_standby_feedback` option"

Reply:
It would also be good to add a link to the GUC. I couldn't find the official Postgres link for it, but https://postgresqlco.nf/doc/en/param/hot_standby_feedback/ might do.